client/core/core.go

```diff
@@ -0,0 +1,251 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package core
+
+import (
+	"context"
+	"encoding/json"
+	"net/url"
+	"sync"
+	"time"
+
+	"decred.org/dcrdex/client/comms"
+	"decred.org/dcrdex/client/db"
+	"decred.org/dcrdex/client/db/bolt"
+	"decred.org/dcrdex/dex"
+	"decred.org/dcrdex/dex/msgjson"
+)
+
+var log dex.Logger
+
+// websocket is satisfied by a comms.WsConn, or a stub for testing.
+type websocket interface {
+	NextID() uint64
+	WaitForShutdown()
+	Send(msg *msgjson.Message) error
+	Request(msg *msgjson.Message, f func(*msgjson.Message)) error
+	MessageSource() <-chan *msgjson.Message
+}
+
+// dexConnection is the websocket connection and the DEX configuration.
+type dexConnection struct {
+	conn   websocket
+	assets map[uint32]*msgjson.Asset
+	cfg    *msgjson.ConfigResult
+}
+
+// Config is the configuration for the Core.
+type Config struct {
+	// DBPath is a filepath to use for the client database. If the database does
+	// not already exist, it will be created.
+	DBPath string
+	// Logger is a logger for the core to use. Having the logger as an argument
+	// enables creating custom loggers for use in a GUI interface.
+	Logger dex.Logger
+	// Certs is a mapping of URL to filepaths of TLS Certificates for the server.
+	// This is intended for accomodating self-signed certificates, mostly for
+	// testing.
```

Comment:
I expect self-signed TLS certs will be used in production for the server's RPC server.  I suppose if the op wants to put up a website they would have certificates signed by a CA, but I wouldn't require that.

Reply:
> server's RPC server

As in the server in **server/comms**? If the server doesn't use a CA, wouldn't they need to supply the certificate to every client? I've always assumed operators would get a domain name and use a CA in production, though I could see cases for small groups sharing a certificate for a private DEX.

Reply:
Right, certificate being provided to the client directly from the server operator, perhaps over a proper web server similar to how https://cspp.decred.org provides the pem cert for the actual CSPP server that listens on a different port.  I don't think I would personally use a certificate signed by a CA for the server/comms.Server TLS RPC server.  @jrick I'm using CSPP as an example, so perhaps you would have an opinion on whether it would make sense to use a CA signed certificate vs. self-signed as our services have done in the past, or even if it's possible or practical to use a CA for this.

Reply:
As one example, how would a CA be usable if you wanted to run a service with just an IP address.